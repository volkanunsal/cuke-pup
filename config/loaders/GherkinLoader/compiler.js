import { createFsFromVolume, Volume } from 'memfs';
import path from 'path';
import webpack from 'webpack';

export default (fixture, options = {}) => {
  const compiler = webpack({
    mode: 'development',
    context: __dirname,
    entry: `./${fixture}`,
    output: {
      path: path.resolve(__dirname),
      filename: 'bundle.js',
    },
    module: {
      rules: [
        {
          test: /\.feature$/,
          use: {
            loader: path.resolve(__dirname, './index.js'),
            options,
          },
        },
      ],
    },
  });

  compiler.outputFileSystem = createFsFromVolume(new Volume());
  compiler.outputFileSystem.join = path.join.bind(path);

  return new Promise((resolve, reject) => {
    compiler.run((err, stats) => {
      if (err) reject(err);
      if (stats.hasErrors()) reject(stats.toJson().errors);

      resolve(stats);
    });
  });
};
