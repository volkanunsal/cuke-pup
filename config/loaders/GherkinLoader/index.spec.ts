import { cucumber } from 'stucumber';
import compiler from './compiler.js';

test('Inserts name and outputs JavaScript', async () => {
  const stats = await compiler('integration.feature');
  const output = stats.toJson({ source: true }).modules[0].source;
  eval(output.toString());

  expect(cucumber.getResults()).toMatchSnapshot();
});
