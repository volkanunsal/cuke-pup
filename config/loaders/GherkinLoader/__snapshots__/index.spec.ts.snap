// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inserts name and outputs JavaScript 1`] = `Array []`;
